web_programming/get_imdbtop.py

```diff
@@ -2,10 +2,68 @@
 from bs4 import BeautifulSoup
 
 
-def imdb_top(imdb_top_n):
+def print_imdb_topn_movies(num_movies: int = 5) -> None:
+    """Crawls IMDB and prints the top n movies with their respective titles, genre,
+    ratings, and imdb page link released.
+
+    Args:
+        num_movies: The number of movies to crawl. Defaults to 5.
+
+    >>> print_imdb_topn_movies()
+    The Shawshank Redemption
```

Comment:
This test result will change if the underlying data changes.

Reply:
Hello, how would you suggest I resolve this? Should I get rid of the doctests in general?

Reply:
The function should return a list or tuple of movies.  The tests should ensure that the length of that list corresponds to `num_movies`.  There should also be tests for zero movies, minus three movies, and 3.2 movies to see how the algorithm deals with garbage input.

Each movie could be a instance of a data class or named tuple.

Reply:
Thank you very much for the detailed explanations. I am working on it!